src/backend/columnar/columnar_tableam.c

```diff
@@ -983,8 +1070,149 @@ columnar_index_build_range_scan(Relation heapRelation,
 								void *callback_state,
 								TableScanDesc scan)
 {
-	ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					errmsg("indexes not supported for columnar tables")));
+	/*
+	 * TODO: Should this function call pgstat_progress_update_param in
+	 * somewhere as heapam_index_build_range_scan ?
+	 */
```

Comment:
consider this

Reply:
Looked into `heapam_index_build_range_scan` a bit more and realized that heapAM actually does some bookkeeping at  `HeapScanDesc` to report how many blocks are read during table scan.

Plus, "block" term doesn't make much sense for columnar tables.
I would vote for skipping progress reporting for now, by converting `TODO` to an `XXX` comment, what do you think @jeff-davis ?